src/mfakto.cpp

```diff
@@ -349,8 +352,16 @@ int init_CL(int num_streams, cl_int *devnumber)
         std::cerr << "Error " << status << " (" << ClErrorString(status) << "): clGetPlatformInfo(VERSION)\n";
         return 1;
       }
-      std::cout << ", Version: " << buf << std::endl;
+      std::cout << ", Version: " << buf << ", Devices: " << device_count << std::endl;
 #endif
+      if (device_count == 0) {
```

Comment:
Doing it here ensures that the "detailed info" code would print all platforms until the end or until the AMD platform is found.